stix_shifter_modules/sentinelone/stix_transmission/ping_connector.py

```diff
@@ -0,0 +1,52 @@
+import json
+from stix_shifter_utils.modules.base.stix_transmission.base_ping_connector \
+    import BasePingConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from requests.exceptions import ConnectionError
+from stix_shifter_utils.utils import logger
+
+
+class PingConnector(BasePingConnector):
+    """Ping connector class """
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: dict
+        """
+        try:
+            # Construct a response object
+            return_obj = {}
+            response_dict = {}
+
+            response = self.api_client.ping_datasource()
+
+            response_code = response.code
+            response_txt = response.read().decode('utf-8')
+
+            response_dict = json.loads(response_txt)
+
+            if response_code == 200:
+                return_obj['success'] = True
+                return_obj['code'] = response_code
+            else:
+                return_obj['success'] = False
+                return_obj['code'] = response_code
+
+        except ConnectionError:
+            response_dict['type'] = "ConnectionError"
+            response_dict['message'] = "Invalid Host"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        except Exception as ex:
+            if 'Max retries exceeded' in str(ex):
```

Comment:
why do you return `success = True` for this exception?

Reply:
The max retries exceeded exception is generated due to api call rate limitation (api supports 1 call per second). Hence considering that data source is active in this exception case for ping.

Reply:
we might hit into false positive issue. `Max retries exceeded` exception may occur for some other reasons as well. what you can do is to send another ping call to the api after 1 second to verify.

Reply:
As per suggestion, now making another ping call after 1 second in case of max retries exceed exception.